python_bindings/src/halide/_generator_helpers.py

```diff
@@ -366,6 +371,22 @@ def _unsorted_cls_dir(cls):
         yield (k, v)
 
 
+_halide_generator_context = ContextVar('halide_generator_context', default=None)
+
+@contextmanager
+def set_context(context:GeneratorContext):
+    token = _halide_generator_context.set(context)
+    try:
+        yield context
+    finally:
+        _halide_generator_context.reset(token)
+
+def _get_generator_context() -> GeneratorContext:
+    context = _halide_generator_context.get()
+    _check(isinstance(context, GeneratorContext), "There is no active GeneratorContext")
```

Comment:
`.get()` above already throws is there is no active context. This check is really only confirming that we didn't internally mess up by calling `set_context` on something wrong.

Reply:
IIUC it should return None:

`_halide_generator_context = ContextVar('halide_generator_context', default=None)`

Reply:
ah you put a default